### PR TITLE
Sandbox: Configurable metrics output.

### PR DESCRIPTION
**Pull request by SamirTalwar-DA**
You can now opt-in to pushing metrics to:

  - the console, with `--metrics-reporter=console`;
  - a directory full of CSV files, with `--metrics-reporter=csv:PATH`; or
  - a Graphite server, with `--metrics-reporter=graphite[:HOST][:PORT]` (defaulting to `localhost:2003`).

You can also customize how often metrics are pushed, with `--metrics-reporting-interval`. This takes a Java Duration, so will probably look like `PT30S`. The default is every 10 seconds.

To make this useful for Sandbox Next, I needed to stop using two separate metrics registries (one for the `StandaloneApiServer`, and one for the `StandaloneIndexerServer`). However, just using a single registry made for a collision in the metrics from the Hikari connection pool. I've namespaced these with a parameter, `name`, because I am boring.

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/metrics/MetricsReporter.scala [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+  case object Console extends MetricsReporter
+
+  final case class Csv(directory: Path) extends MetricsReporter
+
+  final case class Graphite(address: InetSocketAddress) extends MetricsReporter
+
+  object Graphite {
+    lazy val defaultHost: InetAddress = InetAddress.getLoopbackAddress
+    lazy val defaultPort: Int = 2003
```
Comment (stefanobaghino-da) [inline ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/metrics/MetricsReporter.scala]:
```suggestion
    val defaultHost: InetAddress = InetAddress.getLoopbackAddress
    val defaultPort: Int = 2003
```
`object`s are already lazy and these two are very cheap, no reason to have more locks in.

**Comment by SamirTalwar-DA** [comment]
Regarding the `ServerName`: we have two Hikari connection pools, and we need to give them two different names for metrics otherwise one of them will fail to start, and everything will go to hell. So we need to pass in a disambiguating name.

I agree that `ServerName` is a bad choice of name, but I can't think of a better one. Can you?

**Comment by SamirTalwar-DA** [comment]
About documenting this: the options are hidden and we intend to keep them that way for now, so I'm not documenting them or adding them to the changelog. If and when we are happy with them and agree that they're staying put, we can open them up for wider use.

**Comment by stefanobaghino-da** [comment]
> Regarding the `ServerName`: we have two Hikari connection pools, and we need to give them two different names for metrics otherwise one of them will fail to start, and everything will go to hell. So we need to pass in a disambiguating name.
> 
> I agree that `ServerName` is a bad choice of name, but I can't think of a better one. Can you?

The only two names are "ledger-api-server" and "indexer". Rather than a tagged type, I'd say you can make this an enumeration ADT. Probably `ServerRole` could be a good name. And probably the string in it could be called "metricsSuffix" or something like that.

**Comment by SamirTalwar-DA** [comment]
@stefanobaghino-da: There's a third, "sandbox", but I agree with your solution. I'll make the changes. 🙂

**Pull request merged**